### ISSUE TITLE
Fully automate the release process

### ISSUE THREAD

**Issue by pepibumur** [xcode-project-manager/xcodeproj#114]
## Context 🕵️‍♀️
Although some of the release steps are automated, there are still some that have to be done manually. There are Ruby tools out there that we could leverage to automate all the steps.

- [GitHub Changelog](https://github.com/skywinder/github-changelog-generator): Generates the Changelog getting the list of changes since the last tag.
- [GitHub Release](https://github.com/mpalmer/github-release): Create releases on GitHub.

## What 🌱
Fully automate the release process. The steps should be:
- Run all the checks locally (build, test, lint).
- Generate the changelog.
- Update the .podspec with the new version.
- Update the Homebrew formula with the new version.
- Commit and tag with the new version.
- Push and create the release with the generated changelog.
- Close the associated milestone.
- Update the formula URL and hash.
- Commit and push the changes in the formula.

> Note: Once this is done we could automate the other organization repositories.

**Comment by pepibumur** [comment]
Related to what we were discussing yesterday @esttorhe

**Comment by esttorhe** [comment]
Yes!!
We'll have to make a `Heroku` app to get a free dyno and schedule releases on a certain day though :p

I can't think of an easy way to release on schedule using `Github` though

¯\\\_(ツ)\_/¯

**Comment by pepibumur** [comment]
@esttorhe I was thinking more about branching out and letting Travis-CI do its job. I think it's very feasible now that we update the `CHANGELOG` every time we open a new PR. Would you like to tackle it? 😀
